timely_beliefs/beliefs/classes.py

```diff
@@ -276,18 +277,29 @@ def add_to_session(
         """
         # Belief timing is stored as the belief horizon rather than as the belief time
         belief_records = (
-            beliefs_data_frame.convert_index_from_belief_time_to_horizon()
-            .reset_index()
-            .to_dict("records")
+            beliefs_data_frame.convert_index_from_belief_time_to_horizon().reset_index()
         )
-        beliefs = [cls(sensor=beliefs_data_frame.sensor, **d) for d in belief_records]
+        beliefs = [
+            cls(sensor=beliefs_data_frame.sensor, **d)
+            for d in belief_records.to_dict("records")
+        ]
+
         if expunge_session:
             session.expunge_all()
-        if not allow_overwrite:
-            if bulk_save_objects:
-                session.bulk_save_objects(beliefs)
-            else:
-                session.add_all(beliefs)
+
+        if bulk_save_objects:
+            # serialize source and sensor
+            belief_records["source_id"] = belief_records["source"].apply(lambda x: x.id)
+            belief_records["sensor_id"] = belief_records.sensor.id
+            belief_records = belief_records.drop(columns=["source"])
+
+            session.execute(
+                insert(cls)
+                .values(belief_records.to_dict("records"))
+                .on_conflict_do_nothing()
```

Comment:
So this would not overwrite old objects with new ones ... would we maybe prefer the other way around?

Or even more complete, we could check `allow_overwrite` here and decide whether to do this  or [`on_conflict_do _update`](https://docs.sqlalchemy.org/en/20/dialects/postgresql.html#sqlalchemy.dialects.postgresql.Insert.on_conflict_do_update)

It seems in that case we'd need to provide fields that need to be part of the update, but I guess we can use all?

Reply:
Thanks! I've experimented a bit with this option and I found a way to update the `event_value` on conflict. I'll push the commit in a minute :)